test/keywordPreservation.test.ts

```diff
@@ -52,32 +52,31 @@ describe('#Keyword Preservation', () => {
     it('should retrieve all preservable fields from assets tree', () => {
       const fieldsToPreserve = getPreservableFieldsFromAssets(
         {
-          object: {
+          tenant: {
             friendly_name: 'Friendly name ##KEYWORD##',
             notInKeywordMapping: '##NOT_IN_KEYWORD_MAPPING##',
             number: 5,
             boolean: true,
             nested: {
               nestedProperty: 'Nested property ##KEYWORD##',
             },
+            nestedArray: [
+              {
+                name: 'nested-array-item-1',
+                value:
+                  "Even with ##KEYWORD##, this won't get preserved because this nested array item does not have a registered resource identifier",
+              },
+            ],
           },
-          array: [
+          actions: [
             {
-              name: 'array-item-1',
-              nestedArray: [
-                {
-                  name: 'nested-array-item-1',
-                  value: 'Nested array value 1 ##KEYWORD##',
-                },
-                {
-                  name: 'nested-array-item-2',
-                  value: 'Nested array value 2 ##KEYWORD##',
```

Comment:
These nested values are removed because with the addition of the resource-specific identifiers, we necessarily do not know how to compare nested array items for uniqueness.